NitroxServer/GameLogic/Entities/GlobalRootData.cs

```diff
@@ -0,0 +1,50 @@
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using ProtoBufNet;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace NitroxServer.GameLogic.Entities;
+
+[DataContract]
+public class GlobalRootData
+{
+    [DataMember(Order = 1)]
+    public List<GlobalRootEntity> Entities = new();
+
+    [ProtoAfterDeserialization]
+    private void ProtoAfterDeserialization()
+    {
+        foreach (GlobalRootEntity entity in Entities)
+        {
+            EnsureChildrenTransformAreParented(entity);
+        }
+    }
+
+    [OnDeserialized]
+    private void JsonAfterDeserialization(StreamingContext context)
+    {
+        ProtoAfterDeserialization();
+    }
+
+    private static void EnsureChildrenTransformAreParented(WorldEntity entity)
+    {
+        if (entity.Transform == null)
+        {
+            return;
+        }
+        foreach (Entity child in entity.ChildEntities)
+        {
+            if (child is WorldEntity childWE && childWE.Transform != null)
```

Comment:
A GlobalRootEntity's child isn't necessarily a GlobalRootEntity. Instead it could be an InventoryItemEntity or even other things I can't think of right now.